src/main/java/org/checkerframework/checker/startswith/StartsWithTransfer.java

```diff
@@ -0,0 +1,88 @@
+import org.checkerframework.checker.startswith.qual.StartsWithUnknown;
+import com.sun.source.tree.Tree;
+
+
+public class StartsWithTransfer extends CFTransfer{
+
+    /**The {@code java.lang.String#startsWith()} method.*/
+    private final ExecutableElement stringStartsWith;
+
+    private StartsWithAnnotatedTypeFactory aTypeFactory;
+
+    public StartsWithTransfer(CFAnalysis analysis) {
+        super(analysis);
+        this.aTypeFactory = (StartsWithAnnotatedTypeFactory) analysis.getTypeFactory();
+        this.stringStartsWith = TreeUtils.getMethod("java.lang.String", "startsWith", 1,
+                                                                        aTypeFactory.getProcessingEnv());
+    }
+
+    /**For all then branches which have a startsWith condition it annotates the receiver with the same annotation as
+     * the argument of startsWith.
+     */
+    @Override
+    protected void processConditionalPostconditions(MethodInvocationNode node, ExecutableElement methodElement,
+                                                    Tree tree, CFStore thenStore, CFStore elseStore) {
+        Node receiver = node.getTarget().getReceiver();
+        Receiver receiverReceiver = FlowExpressions.internalReprOf(aTypeFactory, receiver);
+        if (methodElement.equals(stringStartsWith)) {
+            AnnotatedTypeMirror atm = aTypeFactory.getAnnotatedType(node.getArgument(0).getTree());
+            thenStore.insertValue(receiverReceiver,
+                                  atm.getAnnotationInHierarchy(aTypeFactory.getCanonicalTopAnnotation()));
+        }
+    }
+
+    /**For all then branches which have a (startsWith && startsWith) condition with the same receivers it annotates the
+     * receiver with the GLB of the annotations of the arguments of startsWith
+     */
+    @Override
+    public TransferResult<CFValue, CFStore> visitConditionalAnd(ConditionalAndNode node,
+                                                                TransferInput<CFValue, CFStore> input){
+        TransferResult<CFValue, CFStore> result = super.visitConditionalAnd(node, input);
+        if(node.getLeftOperand().getTree().getKind() == Tree.Kind.METHOD_INVOCATION &&
+                node.getRightOperand().getTree().getKind() == Tree.Kind.METHOD_INVOCATION){
+            MethodInvocationNode leftNode = (MethodInvocationNode) node.getLeftOperand();
+            MethodInvocationNode rightNode = (MethodInvocationNode) node.getRightOperand();
+            if(leftNode.getTarget().getReceiver().equals(rightNode.getTarget().getReceiver())) {
+                ExecutableElement methodElementLeft = leftNode.getTarget().getMethod();
+                ExecutableElement methodElementRight = rightNode.getTarget().getMethod();
+                if (methodElementLeft.equals(stringStartsWith) && methodElementRight.equals(stringStartsWith)) {
+                    Node receiver = leftNode.getTarget().getReceiver();
```

Comment:
You called `leftNode.getTarget().getReceiver()` already a few lines earlier. I suggest making it a local variable so you don't have to repeat yourself.